src/Chain/InputProcessor/SystemPromptInputProcessor.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Chain\InputProcessor;
+
+use PhpLlm\LlmChain\Chain\Input;
+use PhpLlm\LlmChain\Chain\InputProcessor;
+use PhpLlm\LlmChain\Model\Message\SystemMessage;
+
+final readonly class SystemPromptInputProcessor implements InputProcessor
+{
+    public function __construct(
+        private string $systemPrompt,
+    ) {
+    }
+
+    public function processInput(Input $input): void
+    {
+        $messages = $input->messages;
+
+        if (null === $messages->getSystemMessage()) {
```

Comment:
We should add a log here at least. I also thought about throwing, but this way you are not able to overwrite the system prompt in certain situations